### PR TITLE
Refactored DiffusionScript

### LINKED ISSUE #3905: Split SimArgs in Diffusion simulation
We ought to split `SimArgs` into two parts:

* `SimArgs` which only contains simulation level configuration,
* `NodeArgs` which only contains node level configuration

and then redefine

```haskell
data DiffusionScript = DiffsionScript SimArgs [(NodeArgs, [Command])]
```

### PR DESCRIPTION
**Pull request by bolt12**
# Description

This PR fixes #3905 

# Checklist

- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Commits have useful messages
    - [x] New tests are added if needed and existing tests are updated
    - [ ] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [ ] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [ ] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [x] Self-reviewed the diff
    - [x] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [x] Reviewer requested

### CODE REVIEW

File: ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs [hunk 1]
```diff
@@ -239,35 +257,116 @@ fixupCommands (jn@(JoinNetwork _ _):t) = jn : go jn t
...
+      , naDNSTimeoutScript      = dnsTimeout
+      , naDNSLookupDelayScript  = dnsLookupDelay
+      }
+  where
+    hasActive :: Int -> PeerSelectionTargets -> Bool
+    hasActive minConn (PeerSelectionTargets _ _ _ y) = y > minConn
+
 -- | Multinode Diffusion Simulator Script
 --
 -- List of 'SimArgs'. Each element of the list represents one running node.
```
Comment (coot) [inline ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs:343]: The haddock needs to be updated.

**Review by coot — APPROVED** [review]
LGTM

File: ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs (hunk 1 above)
Comment (bolt12) [inline ouroboros-network/test/Test/Ouroboros/Network/Testnet/Simulation/Node.hs:343]:
```suggestion
-- 'SimArgs' with all the values needed for running the simulation, followed
-- by a list of 'NodeArgs' where each element represents one running node and
-- respective 'Command's.
```

**Comment by bolt12** [comment]
bors merge

**Comment by coot** [comment]
bors merge

**Comment by coot** [comment]
There was a test failure which I recorded in #4092.

**Comment by coot** [comment]
bors merge

**Pull request merged**